src/transformers/trainer_tf.py

```diff
@@ -620,13 +620,23 @@ def apply_gradients(self, features, labels):
             self.optimizer.apply_gradients(list(zip(gradients, self.model.trainable_variables)))
         else:
             for _ in tf.range(self.args.gradient_accumulation_steps):
-                reduced_features = features[: self.args.train_batch_size / self.args.n_replicas]
-                reduced_labels = labels[: self.args.train_batch_size / self.args.n_replicas]
+                reduced_features = {
+                    ft: features[ft][: self.args.train_batch_size // self.args.n_replicas] for ft in features
+                }
+                reduced_labels = labels[: self.args.train_batch_size // self.args.n_replicas]
 
                 self.training_step(reduced_features, reduced_labels)
 
-                features = tf.concat(
-                    [features[self.args.train_batch_size / self.args.n_replicas :], reduced_features], axis=0
+                features = {
+                    ft: tf.concat(
```

Comment:
Same here, I'd loop on `features.items()`

Reply:
Good to me! Doing the update.

Reply:
Does it looks like you proposed now?

Reply:
Yup, thanks (I think this is slightly faster this way.)

Reply:
Perfect, looks good to merge for me then :)